include/bluetooth/mesh/gen_lvl.h

```diff
@@ -108,6 +108,12 @@ struct bt_mesh_lvl_status {
 #define BT_MESH_LVL_MSG_MAXLEN_DELTA_SET 7
 #define BT_MESH_LVL_MSG_MINLEN_MOVE_SET 3
 #define BT_MESH_LVL_MSG_MAXLEN_MOVE_SET 5
+
+#ifdef CONFIG_BT_MESH_SCENE_SRV
+#define BT_MESH_LVL_SCENE_ENTRY_INIT .scene = BT_MESH_SCENE_ENTRY(&_bt_mesh_lvl_srv_scene_type),
```

Comment:
A bit dirty, but I suppose it works :smile: I think I would ditch the trailing comma. 

This is bordering on over-optimization, I think, and just keeping the pointer around even when scene isn't active might be just as good an option at this point. I'd like to hear some other opinions though, e.g. from @alxelax or @ludvigsj. 

I won't hold up this PR for this, though, there are pros and cons to both suggestions. If anyone else feels strongly about it, I'd like to hear it.

Reply:
Kind of arguable improvement. Actually, I like the direction that @PavelVPV chose for optimization but I'd like to see more macro magic. :)
I'd suggest passing a scene instance as a macro parameter always. Provide NULL if we want to use a scene from the extended model.  What I see now @PavelVPV  had to change macros for initialization by manual rolling out of parameters. That ditches the pros of macro presence. 
I'd say it needs a bit more work for further improvement.  

Since we started way of macro magic it is possible of scene pointer declaration to hide under another macro to get rid of ifdef within servers structure. It will make code a bit clearer.

Reply:
If we do so, customers will have to pass scene types (`_bt_mesh_lvl_srv_scene_type`).

Reply:
yep. This is also possible to hide under macro :) My opinion, if we started preprocessor magic it is worth using it as maximum as possible to get a very short final result. But up to you of course. I didn't request changes.

Reply:
```
 include/bluetooth/mesh/gen_lvl_srv.h          |  4 ++--
 include/bluetooth/mesh/lightness_srv.h        | 10 +++++++---
 tests/subsys/bluetooth/mesh/models/prj.conf   |  4 ++++
 tests/subsys/bluetooth/mesh/models/src/main.c |  2 +-
 4 files changed, 14 insertions(+), 6 deletions(-)

diff --git a/include/bluetooth/mesh/gen_lvl_srv.h b/include/bluetooth/mesh/gen_lvl_srv.h
index eb86bce4f..a20763e68 100644
--- a/include/bluetooth/mesh/gen_lvl_srv.h
+++ b/include/bluetooth/mesh/gen_lvl_srv.h
@@ -30,10 +30,10 @@ struct bt_mesh_lvl_srv;
  *
  * @param[in] _handlers Pointer to a handler function structure.
  */
-#define BT_MESH_LVL_SRV_INIT(_handlers)                                        \
+#define BT_MESH_LVL_SRV_INIT(_handlers, ...)                                   \
 	{                                                                      \
 		.handlers = _handlers,                                         \
-		BT_MESH_LVL_SCENE_ENTRY_INIT                                   \
+		__VA_ARGS__                                                    \
 	}
 
 /** @def BT_MESH_MODEL_LVL_SRV
diff --git a/include/bluetooth/mesh/lightness_srv.h b/include/bluetooth/mesh/lightness_srv.h
index 11795c3bd..8f89a032e 100644
--- a/include/bluetooth/mesh/lightness_srv.h
+++ b/include/bluetooth/mesh/lightness_srv.h
@@ -27,6 +27,12 @@ extern "C" {
 
 struct bt_mesh_lightness_srv;
 
+#ifdef CONFIG_BT_MESH_SCENE_SRV
+#define EMPTY_NULL NULL
+#else
+#define EMPTY_NULL
+#endif
+
 /** @def BT_MESH_LIGHTNESS_SRV_INIT
  *
  * @brief Initialization parameters for @ref bt_mesh_lightness_srv.
@@ -35,9 +41,7 @@ struct bt_mesh_lightness_srv;
  */
 #define BT_MESH_LIGHTNESS_SRV_INIT(_handlers)                                  \
 	{                                                                      \
-		.lvl = {                                                       \
-			.handlers = &_bt_mesh_lightness_srv_lvl_handlers,      \
-		},                                                             \
+		BT_MESH_LVL_SRV_INIT(&_bt_mesh_lightness_srv_lvl_handlers),    \
 		.ponoff = {                                                    \
 			.onoff = {                                             \
 				.handlers =                                    \
diff --git a/tests/subsys/bluetooth/mesh/models/prj.conf b/tests/subsys/bluetooth/mesh/models/prj.conf
index 55ead6f05..9c9599b0a 100644
--- a/tests/subsys/bluetooth/mesh/models/prj.conf
+++ b/tests/subsys/bluetooth/mesh/models/prj.conf
@@ -66,3 +66,7 @@ CONFIG_BT_MESH_LIGHT_SAT_SRV=y
 CONFIG_BT_MESH_LIGHT_HSL_SRV=y
 CONFIG_BT_MESH_LIGHT_HSL_CLI=y
 CONFIG_BT_MESH_SCHEDULER_CLI=y
+
+CONFIG_BT_SETTINGS=n
+CONFIG_BT_MESH_SCENE_SRV=n
+CONFIG_BT_MESH_SCHEDULER_SRV=n
diff --git a/tests/subsys/bluetooth/mesh/models/src/main.c b/tests/subsys/bluetooth/mesh/models/src/main.c
index f60b9aa90..5a19688f2 100644
--- a/tests/subsys/bluetooth/mesh/models/src/main.c
+++ b/tests/subsys/bluetooth/mesh/models/src/main.c
@@ -8,7 +8,7 @@
 #include <bluetooth/mesh/models.h>
 
 /* Instantiate all models, using their INIT macros if available. */
-static struct bt_mesh_lvl_srv lvl_srv = BT_MESH_LVL_SRV_INIT(NULL);
+static struct bt_mesh_lvl_srv lvl_srv = BT_MESH_LVL_SRV_INIT(NULL, BT_MESH_LVL_SCENE_ENTRY_INIT);
 static struct bt_mesh_ponoff_srv ponoff_srv =
 	BT_MESH_PONOFF_SRV_INIT(NULL, NULL, NULL);
 static struct bt_mesh_plvl_srv plvl_srv = BT_MESH_PLVL_SRV_INIT(NULL);
 ```

Reply:
this is my experiment how to add more preprocessor magic to hide a little bit more internal stuff and to be able to reuse INIT macros

Reply:
This introduces some boilerplate code for customers. On the other hand, this can help to remove scene entry for Light Lightness server when it is coupled with Light LC server. But this needs to be clarified in the documentation. What do you think @trond-snekvik ?